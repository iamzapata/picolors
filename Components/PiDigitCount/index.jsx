import React from 'react'
import { charOcurrenceCount, piDecimals } from 'utils'
import GridStyles from 'styles/grids/Grid_10_Items.module.scss'
import Styles from 'Components/PiDigitCount/PiDigitCount.module.css'

import classNames from 'classnames'

const PiDigitCount = () => (
  <div className="m-3">
    <h6 className="text-center mb-1">Ocurrences</h6>
    <div className={classNames(GridStyles.Grid_3_3)}>
      {[0, 1, 2, 3, 4, 5, 6, 7, 8, 9].map((index) => {
        return (
          <div key={index} className={Styles.DigitBox}>
            <div className="fw-bold text-center">{index}</div>

            <span key={index} className="small text-center">
              {new Intl.NumberFormat('en-US', {
                maximumSignificantDigits: 8,
              }).format(charOcurrenceCount(piDecimals, index))}
            </span>
          </div>
        )
      })}
    </div>
  </div>
)

export default PiDigitCount
