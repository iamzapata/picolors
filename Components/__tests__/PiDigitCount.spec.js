import { render } from '@testing-library/react'
import PiDigitCount from 'Components/PiDigitCount'

describe('<PiDigitCount/>', () => {
  it('renders', () => {
    const { container } = render(<PiDigitCount />)
    expect(container).toMatchSnapshot()
  })
})
