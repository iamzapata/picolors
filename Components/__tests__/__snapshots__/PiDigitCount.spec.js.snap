// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PiDigitCount/> renders 1`] = `
<div>
  <div
    class="m-3"
  >
    <h6
      class="text-center mb-1"
    >
      Ocurrences
    </h6>
    <div
      class=""
    >
      <div>
        <div
          class="fw-bold text-center"
        >
          0
        </div>
        <span
          class="small text-center"
        >
          99,959
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          1
        </div>
        <span
          class="small text-center"
        >
          99,758
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          2
        </div>
        <span
          class="small text-center"
        >
          100,026
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          3
        </div>
        <span
          class="small text-center"
        >
          100,229
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          4
        </div>
        <span
          class="small text-center"
        >
          100,230
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          5
        </div>
        <span
          class="small text-center"
        >
          100,359
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          6
        </div>
        <span
          class="small text-center"
        >
          99,548
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          7
        </div>
        <span
          class="small text-center"
        >
          99,800
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          8
        </div>
        <span
          class="small text-center"
        >
          99,985
        </span>
      </div>
      <div>
        <div
          class="fw-bold text-center"
        >
          9
        </div>
        <span
          class="small text-center"
        >
          100,106
        </span>
      </div>
    </div>
  </div>
</div>
`;
